### PR TITLE
Fetch policy links hash from publishing api instead of storing in the database

### PR DESCRIPTION
**Pull request by MatMoore**
Remove links data from the database and use publishing API v2 instead. This is so that the publishing API becomes the single source of truth for this data, and we don't run into conflicts when links are updated from different places.

Trello ticket https://trello.com/c/BQDQ824W 

This is an updated pull request containing the changes from https://github.com/alphagov/policy-publisher/pull/121 - which in turned contained @tijmenb's work on the ticket.

The policy controller has been refactored to introduce a new form object, which is responsible for all validation relating to the links. We model the links differently when presenting it to the user, so the form also deals with this transformation.

In the content schema, we model it as organisations and lead_organisations (a subset of organisations).

To the user, we separate it out into lead and supporting (disjoint sets) to match whitehall.

There is a commit here to introduce some object level caching for data returned by the publishing-api. This was because these calls were being issued multiple times and loading/submitting the edit page was painfully slow. This is still a bit slower than we'd like, so we may want to follow this up in a different ticket.

Previously the policy publisher maintained the ordering of each of the links, so that users could emphasise certain organisations by placing them first. This is no longer possible because publishing-api doesn't care about the publishing order, and instead orders things alphabetically. What we've done instead is introduce a new link type (lead-organisations) to distinguish between "lead" and "supporting" organisations.

This PR includes a database migration to remove from the schema. We will need to write a separate rake task to populate lead-organisations within the publishing-api database, and we may want to merge that first depending on where that lives.

### CODE REVIEW

**Comment by MatMoore** [comment]
We need to merge https://github.com/alphagov/govuk-content-schemas/pull/213 before the tests will pass for this one, but everything is passing locally.

**Comment by rboulton** [comment]
alphagov/govuk-content-schemas#213 merged

File: db/migrate/20160114121308_remove_link_fields.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class RemoveLinkFields < ActiveRecord::Migration
```
Comment (rboulton) [inline db/migrate/20160114121308_remove_link_fields.rb]: I think we should do this migration in a separate PR, but add in this PR a script which copies the data from these link fields into publishing-API.  Then we can deploy this PR, copy the data across to publishing API, and later remove the old data once everything is working nicely.

Comment (MatMoore) [inline db/migrate/20160114121308_remove_link_fields.rb]: I was thinking we could do that script in a separate PR and deploy that one first, rather than splitting this one up. Would that be simpler?

We haven't got around to writing the script itself yet, but @Davidslv knows about it.

Comment (rboulton) [inline db/migrate/20160114121308_remove_link_fields.rb]: That seems fine, though we'd ideally like to ensure we can deploy this smoothly.  We don't have much tagging at all going on at present, though, so as long as we can copy the data over, then deploy this change within a few minutes, I think it will be fine.

Comment (jackscotti) [inline db/migrate/20160114121308_remove_link_fields.rb]: The script is here: https://github.com/alphagov/policy-publisher/pull/124

Can I remove this migration from this pull request?

Comment (rboulton) [inline db/migrate/20160114121308_remove_link_fields.rb]: Makes sense to remove the migration - happy to merge other than that.

**Comment by jackscotti** [comment]
@rboulton migration removed.

File: db/schema.rb [hunk 2]
```diff
@@ -34,6 +32,8 @@
     t.datetime "created_at",                                   null: false
     t.datetime "updated_at",                                   null: false
     t.string   "email_alert_signup_content_id"
+    t.text     "organisation_content_ids",      default: [],                array: true
```
Comment (mobaig) [inline db/schema.rb]: is this right? thought we dropped these columns?

Comment (jackscotti) [inline db/schema.rb]: Dropped in a previous commit of this PR

https://github.com/alphagov/policy-publisher/commit/5ff57d45e967bc18225b3fbaa622f38ad842e0b4#diff-8a1923c466f71d93472de30d1c687a83R4

This commits puts it back in.
